gateway/src/apicast/policy/nginx_metrics/nginx_metrics.lua

```diff
@@ -112,8 +118,17 @@ function _M:metrics()
   end
 end
 
+local function report_req_response_time()
+  local resp_time = tonumber(ngx.var.request_time)
+
+  if resp_time and response_times then
+    response_times:observe(resp_time)
+  end
+end
+
 function _M.log()
   upstream_metrics.report(ngx.var.upstream_status, ngx.var.upstream_response_time)
+  report_req_response_time()
```

Comment:
Does this include post action time? I guess it does right? Then it might not be exactly correct, because client got the response before post action finished.